nx/lib/nx/lin_alg.ex

```diff
@@ -967,4 +967,96 @@ defmodule Nx.LinAlg do
       opts
     )
   end
+
+  @doc """
+  Produces the tensor taken to the given power by dot-product.
+
+  The input is always a square tensor and a non-negative integer,
+  and the output is a square tensor of the same dimensions as the input tensor.
+
+  The dot-products are unrolled inside `defn`.
+
+  ## Examples
+
+      iex> Nx.LinAlg.dot_power(Nx.tensor([[1, 2], [3, 4]]), 0)
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [1, 0],
+          [0, 1]
+        ]
+      >
+
+      iex> Nx.LinAlg.dot_power(Nx.tensor([[1, 2], [3, 4]]), 6)
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [5743, 8370],
+          [12555, 18298]
+        ]
+      >
+
+      iex> Nx.LinAlg.dot_power(Nx.eye(3), 65535)
+      #Nx.Tensor<
+        s64[3][3]
+        [
+          [1, 0, 0],
+          [0, 1, 0],
+          [0, 0, 1]
+        ]
+      >
+
+      iex> Nx.LinAlg.dot_power(Nx.tensor([[1, 2], [3, 4]]), -1)
+      #Nx.Tensor<
+        f32[2][2]
+        [
+          [-2.0, 1.0],
+          [1.5, -0.5]
+        ]
+      >
+
+      iex> Nx.LinAlg.dot_power(Nx.tensor([[1, 2], [3, 4], [5, 6]]), 1)
+      ** (ArgumentError) expected tensor to match shape {x, x}, got tensor with shape {3, 2}
+  """
+  @doc from_backend: false
+  def dot_power(tensor, power) when is_integer(power) and power < 0 do
+    dot_power(Nx.LinAlg.invert(tensor), -power)
```

Comment:
```suggestion
    tensor |> invert() |> dot_power(abs(power))
```

```suggestion
    dot_power(invert(tensor), abs(power))
```

I know `- power when power < 0` is equivalent to `abs`, but I think `abs` reduces the cognitive overload for the code here. I don't have a strong opinion on whether to use the pipe here, though.